app/unit-tests/src/org/commcare/android/tests/processing/QuarantinedFormsTest.java

```diff
@@ -0,0 +1,31 @@
+package org.commcare.android.tests.processing;
+
+import org.commcare.utils.FormUploadUtil;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+/**
+ * Created by amstone326 on 6/14/17.
+ */
+
+public class QuarantinedFormsTest {
```

Comment:
We have mock response endpoints for data restores, it might be worth seeing how hard it would be to add mock endpoints for submissions.

Here's the restore Mock: https://github.com/dimagi/commcare-android/blob/master/app/unit-tests/src/org/commcare/android/mocks/ModernHttpRequesterMock.java

I'm a bit concerned that a calabash test her might be pretty fragile, but since you've worked here more recently than I mean I could be wrong.